### PR TITLE
feat!: removed require_ssl and change max provider to 6.X

### LINKED ISSUE #641: Google Terraform provider support version 6.x.x+
### TL;DR

The Google Terraform Provider released version 6.0.0 on August 26; https://github.com/hashicorp/terraform-provider-google/releases/tag/v6.0.0.
It would be nice if this module supported version 6.x.x.

### Terraform Resources

_No response_

### Detailed design

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by imrannayer**
removed `settings.ip_configuration.require_ssl` in `google_sql_database_instance`. Please use `settings.ip_configuration.ssl_mode` instead.

fix #641

### CODE REVIEW

**Comment by imrannayer** [comment]
@bharathkkb plz review

File: docs/upgrading_to_sql_db_22.0.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Upgrading to SQL DB 22.0
+
+The 22.0 release of SQL DB is a backward incompatible release.
+
+# Maximum provider version
+This update requires upgrading the minimum Terraform version `1.3`. Maximum provider version is relaxed to use provider version 6.X+
+
+# Removed settings.ip_configuration.require_ssl
+Removed `settings.ip_configuration.require_ssl` from all the modules (`google_sql_database_instance`) in favor of `settings.ip_configuration.ssl_mode`. This field is not available in [provider version 6+](https://github.com/hashicorp/terraform-provider-google/pull/19263)
```
Comment (bharathkkb) [inline docs/upgrading_to_sql_db_22.0.md:9]:
```suggestion
Removed `settings.ip_configuration.require_ssl` from all the modules (`google_sql_database_instance`) in favor of `settings.ip_configuration.ssl_mode`. This field is not available in [provider version 6+](https://registry.terraform.io/providers/hashicorp/google/5.43.0/docs/guides/version_6_upgrade#resource-google_sql_database_instance)
```

File: test/integration/mssql-ha/mssql_ha_test.go [hunk 2]
```diff
@@ -58,7 +58,6 @@ func TestMsSqlHaModule(t *testing.T) {
 		// assert network settings
 		authNetworks := op.Get("settings.ipConfiguration.authorizedNetworks").Array()
 		assert.Equal(1, len(authNetworks), "Expected one auth network")
-		assert.True(op.Get("settings.ipConfiguration.requireSsl").Bool(), "SSL is required")
```
Comment (bharathkkb) [inline test/integration/mssql-ha/mssql_ha_test.go:61]: Could we swap this with the new ssl_mode rather than removing the check?

**Review by bharathkkb — APPROVED** [review]
Thanks @imrannayer, some minor comments

**Pull request merged**